cfme/infrastructure/host.py

```diff
@@ -86,11 +87,22 @@
 pol_btn = partial(tb.select, 'Policy')
 pow_btn = partial(tb.select, 'Power')
 lif_btn = partial(tb.select, 'Lifecycle')
+mon_btn = partial(tb.select, 'Monitoring')
+
 
 match_page = partial(match_location, controller='host',
                      title='Hosts')
 
 
+class InfraHostTimelinesView(TimelinesView, cfme.BaseLoggedInPage):
+
+    @property
+    def is_displayed(self):
+        return all((self.logged_in_as_current_user,
+                   self.navigation.currently_selected == ['Compute', 'Infrastructure', '/host'],
+                   TimelinesView.is_displayed))
```

Comment:
So, I advocate against all() because for it to work, all args need to be evaluated whereas and will False after first False (this is potentially slower).

And the TimelinesView - I think you want to do something like super(TimelinesView, self).is_displayed.